Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+
+            string jsonChunk = Encoding.ASCII.GetString(glbData, stride * 5, (int)chunk0Length);
+
+            var gltfObject = GetGltfObjectFromJson(jsonChunk);
+
+            var chunk1Length = BitConverter.ToUInt32(glbData, stride * 5 + (int)chunk0Length);
+            var chunk1Type = BitConverter.ToUInt32(glbData, stride * 6 + (int)chunk0Length);
+
+            if (chunk1Type != (ulong)GltfChunkType.BIN)
+            {
+                Debug.LogError("Expected chunk 1 to be BIN data!");
+                return null;
+            }
+
+            Debug.Assert(gltfObject.buffers[0].byteLength == chunk1Length, "chunk 1 & buffer 0 length mismatch");
+
+            gltfObject.buffers[0].BufferData = new byte[(int)chunk1Length];
+            Array.Copy(glbData, stride * 7 + (int)chunk0Length, gltfObject.buffers[0].BufferData, 0, (int)chunk1Length);
+
+            return gltfObject;
+        }
+
+        /// <summary>
+        /// Get a single Json Object using the handle provided.
+        /// </summary>
+        /// <param name="jsonString">The json string to search.</param>
+        /// <param name="handle">The handle to look for.</param>
+        /// <returns>A snippet of the json string that defines the object.</returns>
+        private static string GetJsonObject(string jsonString, string handle)
+        {
+            var regex = new Regex($"\"{handle}\"\\s*:\\s*\\{{");
+            var match = regex.Match(jsonString);
+            return match.Success ? GetJsonObject(jsonString, match.Index + match.Length) : null;
+        }
+
+        private static List<string> GetGltfMeshPrimitiveAttributes(string jsonString)
+        {
+            var regex = new Regex("(?<Attributes>\"attributes\"[^}]+})");
+            return GetGltfMeshPrimitiveAttributes(jsonString, regex);
+        }
+
+        private static List<string> GetGltfMeshPrimitiveAttributes(string jsonString, Regex regex)
+        {
+            var jsonObjects = new List<string>();
+
+            if (!regex.IsMatch(jsonString))
+            {
+                return jsonObjects;
+            }
+
+            MatchCollection matches = regex.Matches(jsonString);
+
+            for (var i = 0; i < matches.Count; i++)
+            {
+                jsonObjects.Add(matches[i].Groups["Attributes"].Captures[0].Value.Replace("\"attributes\":", string.Empty));
+            }
+
+            return jsonObjects;
+        }
+
+        /// <summary>
+        /// Get a collection of glTF Extensions using the handle provided.
+        /// </summary>
+        /// <param name="jsonString">The json string to search.</param>
+        /// <param name="handle">The handle to look for.</param>
+        /// <returns>A collection of snippets with the json string that defines the object.</returns>
+        private static Dictionary<string, string> GetGltfExtensionObjects(string jsonString, string handle)
+        {
+            // Bug: sometimes name isn't always before extension declaration
```

Comment:
what frequency have you seen this as an issue?

Reply:
It's more or less a problem with some exporters that don't adhere to the glTF schema.

Reply:
Do we generate a valid game object if extensions are declared before the name? It's probably worthwhile calling something out here that the name isn't always before the extension/developers could encounter issues, but we may as well provide more information/avoid making things look like a bug in the MRTK if they're a bug in the exporter:

// Assumption: This code assumes that a name is declared before extensions in the glTF schema. This may not work for all exporters. Some may fail to adhere to the glTF schema.

Reply:
picking up this fix